File: crates/iota-sdk/examples/epoch.rs
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use iota_sdk::graphql_client::{Client, error::GraphQLResult};

#[tokio::main]
async fn main() -> GraphQLResult<()> {
    let client = Client::new_testnet();

    let current_epoch = client.epoch(None).await?.unwrap();
    println!("Current epoch: {}", current_epoch.epoch_id);
    println!(
        "Current epoch start time: {}",
        current_epoch.start_timestamp.0
    );

    let previous_epoch = client
        .epoch(Some(current_epoch.epoch_id - 1))
        .await?
        .unwrap();
    println!("Previous epoch: {}", previous_epoch.epoch_id);
    println!(
        "Previous epoch stake rewards: {}",
        previous_epoch.total_stake_rewards.unwrap().0
    );

    Ok(())
}

File: bindings/wasm/examples/epoch.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient, initAsync } from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newTestnet();

const currentEpoch = await client.epoch();
if (currentEpoch === null) {
  throw new Error("missing current epoch");
}

console.log(`Current epoch: ${currentEpoch.epochId}`);
console.log(`Current epoch start time: ${currentEpoch.startTimestamp}`);

const previousEpochId = currentEpoch.epochId - 1n;
const previousEpoch = await client.epoch(previousEpochId);
if (previousEpoch === null) {
  throw new Error("missing previous epoch");
}

console.log(`Previous epoch: ${previousEpoch.epochId}`);
if (previousEpoch.totalStakeRewards !== null) {
  console.log(
    `Previous epoch stake rewards: ${previousEpoch.totalStakeRewards}`,
  );
}
